pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -1870,3 +1889,273 @@ func uplinkToPhysdev(ifname string) string {
 	log.Functionf("uplinkToPhysdev found %s", kernIfname)
 	return kernIfname
 }
+
+// Network Instance Address Management Module Initialization
+func initializeIPAddrMgmtModule(ctx *zedrouterContext,
+	status *types.NetworkInstanceStatus) {
+	if status.HostIPAddrModInitialized {
+		return
+	}
+	log.Functionf("initializeIPAddrMgmtModule(%s-%s)\n",
+		status.DisplayName, status.UUID.String())
+	if err := doNetworkInstanceStatusDhcpRangeSanityCheck(status); err == nil {
```

Comment:
I don't think we need these checks. The key check (which is actually missing in the original code) is in the lookupOrAllocateIPv4 where we should have a check that we don't run outside of dhcp.end or the end of the subnet (if dhcp.end is not set).

But these functions don't log any errors; just set the HostIPAddressCount which we can do without (it is likely to be misleading when there is a mixture of static and dynamic allocations in any case).

Reply:
Sure, willl rework this portion of changes. And added the range checks inside AllocateIPv4,